### ISSUE TITLE
Issue with install pytest-dependency

### ISSUE THREAD

**Issue by abotalov** [kennethreitz/pipenv#878]
When I run `pipenv install` for a Pipfile that contains `pytest-dependency` package I get the following error:
```
Traceback (most recent call last):
  File "C:\Python36\Scripts\pipenv-script.py", line 11, in <module>
    load_entry_point('pipenv==8.2.6', 'console_scripts', 'pipenv')()
  File "c:\python36\lib\site-packages\pipenv\vendor\click\core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "c:\python36\lib\site-packages\pipenv\vendor\click\core.py", line 697, in main
    rv = self.invoke(ctx)
  File "c:\python36\lib\site-packages\pipenv\vendor\click\core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python36\lib\site-packages\pipenv\vendor\click\core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python36\lib\site-packages\pipenv\vendor\click\core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "c:\python36\lib\site-packages\pipenv\cli.py", line 1785, in install
    do_init(dev=dev, allow_global=system, ignore_pipfile=ignore_pipfile, system=system, skip_lock=skip_lock, verbose=verbose
, concurrent=concurrent, deploy=deploy)
  File "c:\python36\lib\site-packages\pipenv\cli.py", line 1293, in do_init
    do_lock(system=system)
  File "c:\python36\lib\site-packages\pipenv\cli.py", line 1083, in do_lock
    pre=pre
  File "c:\python36\lib\site-packages\pipenv\utils.py", line 421, in resolve_deps
    resolved_tree.update(resolver.resolve(max_rounds=PIPENV_MAX_ROUNDS))
  File "c:\python36\lib\site-packages\pipenv\patched\piptools\resolver.py", line 101, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "c:\python36\lib\site-packages\pipenv\patched\piptools\resolver.py", line 199, in _resolve_one_round
    for dep in self._iter_dependencies(best_match):
  File "c:\python36\lib\site-packages\pipenv\patched\piptools\resolver.py", line 293, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "c:\python36\lib\site-packages\pipenv\patched\piptools\repositories\pypi.py", line 171, in get_dependencies
    result = reqset._prepare_file(self.finder, ireq)
  File "c:\python36\lib\site-packages\pipenv\patched\pip\req\req_set.py", line 639, in _prepare_file
    abstract_dist.prep_for_dist()
  File "c:\python36\lib\site-packages\pipenv\patched\pip\req\req_set.py", line 134, in prep_for_dist
    self.req_to_install.run_egg_info()
  File "c:\python36\lib\site-packages\pipenv\patched\pip\req\req_install.py", line 438, in run_egg_info
    command_desc='python setup.py egg_info')
  File "c:\python36\lib\site-packages\pipenv\patched\pip\utils\__init__.py", line 707, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in C:\Users\my_username\App
Data\Local\Temp\tmprtpkxb6bbuild\pytest-dependency\
```

Python version: 3.6.3
Pipenv version: 8.2.6
Pip version: 9.0.1

**Comment by paulochf** [comment]
Same here, but using Python 3.6.1 while trying to install matplotlib.

**Comment by vphilippon** [comment]
For `pytest-dependency`, it looks like it need `pytest` to be **already installed** in order to be able to build (the setup.py imports a module that imports pytest). That's not cool. That is why we get an error when trying to build.
That's an issue with the `pytest-dependency` package itself, there's not much to be done here IMHO.
The workaround: Always install `pytest` manually before installing `pytest-dependency`.

For `matplotlib`, I've messed around a bit and noticed that `subprocess32`. a py2 only dependency, is added while it should not, which causes the issue. I'm pretty sure it's fixed by #833, which should land when 8.2.3 gets released.

**Comment by nateprewitt** [comment]
Thanks for all of your research @vphillippon! Yes, both of these issues are related to poorly written setup.py files by the distributors of these packages. Unfortunately, there isn't a lot we can do from our end. My only advice would be to prompt the package owner to determine a fix on their end.

**Comment by vphilippon** [comment]
For the record, with `matplotlib`, it's actually an issue with `pipenv`/`piptools` (fixed by #833, but not yet released) wrongly getting the dependencies from any wheel for any OS/python version, not sticking to the one for the current platform, ending up fetching some py2 dependencies even on py3. `matplotlib` did nothing wrong (even if their `setup.py` is beefy, the dependencies are allright).

Just to not blame the author of `matplotlib` here 😄
